components/DeleteModal.tsx

```diff
@@ -1,36 +1,39 @@
-import { Dispatch, SetStateAction } from 'react';
 import { Gift, QueryKeys } from '~/shared/types';
 import { Modal } from './Modal';
 import { Button } from './Button';
 import { deleteGift } from '~/utils/apiRequests';
-import { handleError } from '~/utils/handleError';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import SvgSpinner from '~/icons/spinner';
+import { useShowErrorToast } from '~/hooks/useShowErrorToast';
 import { handleErrorToast } from '~/utils/handleToasts';
-import { useQueryClient } from '@tanstack/react-query';
+import { handleError } from '~/utils/handleError';
 
 type DeleteModal = {
   gift: Gift;
-  setIsModalOpen: Dispatch<SetStateAction<boolean>>;
+  closeModal: () => void;
 };
 
-export function DeleteModal({ gift, setIsModalOpen }: DeleteModal) {
+export function DeleteModal({ gift, closeModal }: DeleteModal) {
   const queryClient = useQueryClient();
 
-  async function handleDeletion() {
-    try {
-      await deleteGift(gift.uuid);
-      setIsModalOpen(false);
-    } catch (e) {
-      handleErrorToast(handleError(e));
-    }
-    await queryClient.invalidateQueries({
-      queryKey: QueryKeys.GIFTS,
-    });
-  }
+  const { mutateAsync, error, isPending } = useMutation({
+    mutationKey: QueryKeys.DELETE_GIFT,
+    mutationFn: async () => await deleteGift(gift.uuid),
+    onSuccess: async () => {
+      // refresh gift list
+      await queryClient.invalidateQueries({
+        queryKey: QueryKeys.GIFTS,
+      });
+      closeModal();
+    },
+  });
+
+  useShowErrorToast(error);
 
   return (
     <Modal
       className="max-w-80"
-      closeModal={() => setIsModalOpen(false)}
+      closeModal={() => closeModal()}
```

Comment:
Tämän voi lyhentää seuraavasti.
```suggestion
      closeModal={closeModal}
```